ios/FluentUI/BottomSheet/BottomSheetViewController.swift

```diff
@@ -0,0 +1,392 @@
+                currentBottomSheetViewController.currentState = futureState
+                currentBottomSheetViewController.view.layoutIfNeeded()
+
+                // when the bottomsheet drawer is expanded, we don't want the UIViews behind the sheet to be accessible.
+                currentBottomSheetViewController.view.accessibilityViewIsModal = futureState == .expand
+                currentBottomSheetViewController.updateResizingHandleViewAccessibility()
+                UIAccessibility.post(notification: .layoutChanged, argument: currentBottomSheetViewController.resizingHandleView)
+
+                completion?()
+                if futureState == .expand {
+                    currentBottomSheetViewController.delegate?.bottomSheetViewControllerDidExpand?(currentBottomSheetViewController)
+                } else {
+                    currentBottomSheetViewController.removeDimmingView()
+                    currentBottomSheetViewController.delegate?.bottomSheetViewControllerDidCollapse?(currentBottomSheetViewController)
+                }
+            }
+        }
+    }
+
+// MARK: pan gesture utilities
+
+    @objc private func handlePan(gesture: UIPanGestureRecognizer) {
+        let positionInSuperview = gesture.translation(in: view.superview)
+
+        switch gesture.state {
+        case .began:
+            panningStart()
+        case .changed:
+            panningProgress(in: positionInSuperview.y)
+        case .ended:
+            let velocity = gesture.velocity(in: view)
+            panningStop(in: positionInSuperview.y, velocity: velocity.y)
+        default:
+           break
+        }
+    }
+
+    private func panningStart() {
+        if animator?.isRunning ?? false {
+           return
+        }
+
+        setupAnimator()
+    }
+
+    private func panningStop(in verticalPos: CGFloat, velocity: CGFloat) {
+        panGestureRecognizer.isEnabled = false
+
+        let windowHeight = view.window?.frame.height ?? 0
+        let shouldReverse: Bool
+        let futureState: BottomSheetViewState
+        switch currentState {
+        case .expand:
+            if verticalPos >= windowHeight / 2 || velocity > Constants.velocityThreshold {
+                shouldReverse = false
+                futureState = .collapse
+            } else {
+                shouldReverse = true
+                futureState = .expand
+            }
+        case .collapse:
+            if verticalPos <= -windowHeight / 2 || velocity <= -1 * Constants.velocityThreshold {
+                shouldReverse = false
+                futureState = .expand
+            } else {
+                shouldReverse = true
+                futureState = .collapse
+            }
+        }
+
+        animator?.isReversed = shouldReverse
+        addAnimatorCompletion(to: futureState, completion: nil)
+        animator?.continueAnimation(withTimingParameters: nil, durationFactor: 0)
+    }
+
+    private func panningProgress(in verticalPos: CGFloat) {
```

Comment:
4 topics I wanted to bring up for discussion:

**1.  Content scroll view interaction while the bottomsheet is being resized:**
There is an issue with the content view while the bottomsheet is being resized (during the animation). The icons are changing sizes and the scrollview is also tracking the gesture and being scrolled.

**2. Is the scrollview (content) supposed to support horizontal scrolling too?**
Related to the item above, I ask this question because if we're not passing the gesture to the content view during the scroll gesture, we need to provide a way for the calling code to allow that on a horizontal/vertical basis.

**3. Safe area insets support?**
In the horizontal orientation (video below) the notch covers the rounded corners and potentially one of the options of the sheet. Is there any definition from design on whether we should support both scenarios (respecting or not the safe areas) or provide a setting for that?

**4. Do we intend to provide the resistance bounce when the user tries to dismiss the bottom sheet?**
This is more of an UX consideration, but I think that the resistance effect when the user tries to drag the bottom sheet down to dismiss it is a great clue that the sheet does not allow dismissing.

https://user-images.githubusercontent.com/68076145/113758852-3d8a5180-96c9-11eb-8008-05f9e9f5c82c.mov

Reply:
1. I think it is more of the details of the collectionviewcontroller setup. I think we are still in discussion if the collection view be part of fluent control or not.
2. as far as I know, no horizontal scrolling is not supported and that is by design
3. no, detail design on safe area so I was making my best guess.
4. don't think that was considered by design but could bring it up to discussion.